src/actions/SmartUpdateAction.php

```diff
@@ -81,7 +81,10 @@ public function getDataProvider()
                 throw new BadRequestHttpException('Where condition is empty!');
             }
 
-            $this->dataProvider->query->andFilterWhere($this->findOptions);
+            $this->dataProvider->query
+                ->andFilterWhere($this->findOptions)
+                ->andWhere(['show_deleted' => 1])
```

Comment:
This change is too widespread — it does not seem to be the right place.

Modify the specific actions of the controllers, where this behavior is required. See example https://github.com/hiqdev/hipanel-module-finance/blob/master/src/controllers/TargetController.php#L75-75